drivers/timer/intel_adsp_timer.c

```diff
@@ -217,13 +226,23 @@ void smp_timer_init(void)
 static int sys_clock_driver_init(void)
 {
 	uint64_t curr = count();
-
-	IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
+	if (!sys_clock_driver_reinit)
+		IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
 	set_compare(curr + CYC_PER_TICK);
 	last_count = curr;
 	irq_init();
 	return 0;
 }
 
-SYS_INIT(sys_clock_driver_init, PRE_KERNEL_2,
-	 CONFIG_SYSTEM_CLOCK_INIT_PRIORITY);
+#ifdef CONFIG_PM_DEVICE
```

Comment:
This does not seem to be right. This function is being called from system power management that only requires `CONFIG_PM` enabled.

Reply:
Good catch, thanks. I've tried to make it using [Device Power Management](https://docs.zephyrproject.org/3.2.0/services/pm/device.html#device-model-with-power-management-support) but dropped this idea. Forgot to adjust #ifdefs. I'll fix it in a moment,